pkg/pillar/conntester/zedcloud.go

```diff
@@ -111,6 +112,13 @@ func (t *ZedcloudConnectivityTester) TestConnectivity(
 				Endpoint:   serverNameAndPort,
 				WrappedErr: err,
 			}
+		} else if portsNotReady := t.getPortsNotReady(err, dns); len(portsNotReady) > 0 {
+			// At least one of the uplink ports is not ready in terms of L3 connectivity.
```

Comment:
What is the effect when one is ready and other mgmt port is not ready?

It might be more predictable to only wait when none of the mgmt ports are ready.

Reply:
If any of the mgmt ports is not ready we will wait. I think this is the right way because fallback to the previous DPC should be something we avoid unless it is absolutely necessary. So if we have for example `eth0` and `wwan0`, then I think it is wise to wait for both port ports to "get ready" (with a time limit), re-test and only then potentially revert the network config if neither works.

Reply:
Also in the testing itself we consider DPC as working if at least one mgmt port provides connectivity to the cloud. So this is in line with that rationale.

Reply:
If eth0 and wwan0 are both management ports and it takes wwan0 a few minutes (connect to cellular network, DHCP, etc) before it has a DNS server, will this "wait for all" slow things down? Will the DomainNetworkStatus have Testing=true for that amount of time, and if it does what are the implications?

Reply:
This extra check is just a continuation of the IP-DNS check (DPCStateIPDNSWait). The same deadline applies (which 2.5 minutes after receiving DPC and commencing verification, i.e. maximum waiting time is not extended), and it is only executed if verification keeps failing and it is due to a missing IP address or DNS config (as confirmed by returned error value).
So in `eth0` + `wwan0` scenario, in most cases `eth0` will be ready first and the test will succeed (marking Testing as false). If `eth0` cannot get IP configuration from DHCP, then we will wait (at most 2.5 minutes) for wwan0 to connect and get config, but this already happens even without this patch (if no port has IP/DNS we wait).
The scenario that I'm changing the behavior of, is if `eth0` gets IP/DNS but testing keeps failing, so we wait (up to those 2.5 minutes) to give other ports a chance for working connectivity. However, in most cases a failure of connectivity test with IP/DNS config available would be transient (for up to 5 seconds) and due to that delay in reloading resolv.conf by the Golang resolver. If it is not transient, i.e. `eth0` has IP/DNS but there really is no connectivity to the controller, wouldn't it make sense to give those 2.5min for wwan0 port to try to connect? After all, the point of having two or more mgmt ports is redundancy, so we shouldn't just hastily mark DPC as faulty if one port is failing and fallback to the previous config.

Reply:
Sure, we shouldn't mark it as failed, but we should mark it as success once one of the ports succeeds. Wasn't obvious to me that would happen but now you've clarified it.